py/client/pydeephaven/session.py

```diff
@@ -734,3 +737,60 @@ def plugin_client(self, exportable_obj: ticket_pb2.TypedTicket) -> PluginClient:
 
         Part of the experimental plugin API."""
         return PluginClient(self, exportable_obj)
+
+    def multi_join(self, input: Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]],
```

Comment:
Users have very very strongly wanted the server and client APIs to be the same.  This has a number of deviations compared to the server that I can see at a glance.
1. `multi_join` is a method on session instead of just being a function.
2. `multi_join` is in a different submodule (`session` instead of `table`).
3. The return type is `Table` instead of `MultiJoinTable`.

How similar can the APIs be?

Reply:
Other clients have taken the approach of returning Table for now, instead of MultiJoinTable. MJT is more of a future-looking API, and given that its potential (adding more joins to an existing MJ and thus getting out a new Table) hasn't been realized yet there's no real benefit to building an object type plugin for MJT.

Reply:
I have moved the new method to table.py as a function and have also relocated the `MultiJoinInput` class there as well. 

@chipkent  I totally understand the desire to have the interface be the same for the two APIs, but the client isn't limited to working with just one server and making a 'session' level function a method on the Session object has the benefit of clear isolation/readability etc. So this change has a trade-off which I am quite ambivalent about.